pybigquery/sqlalchemy_bigquery.py

```diff
@@ -363,22 +363,51 @@ def _split_table_name(full_table_name):
             dataset, table_name = table_name_split
         elif len(table_name_split) == 3:
             project, dataset, table_name = table_name_split
+        else:
+            raise ValueError("Did not understand table_name: {}".format(full_table_name))
```

Comment:
I used .format() instead of f-strings since I assume you want to support older Python versions.  The underlying BigQuery libraries seemed to use ValueError for SQL parsing issues, so I this was a good exception type.